honeybee_energy/schedule/ruleset.py

```diff
@@ -1043,6 +1056,150 @@ def extract_all_from_idf_file(idf_file):
+                        return rule
+
+                    # Create rules for regular days
+                    for rule_name in [
+                        "alldays",
+                        "weekdays",
+                        "weekends",
+                        "sunday",
+                        "monday",
+                        "tuesday",
+                        "wednesday",
+                        "thursday",
+                        "friday",
+                        "saturday",
+                    ]:
+                        if rule_name in field:
+                            rule = create_rule_for(rule_name)
+                            rules.append(rule)
+
+                    # Create rules for holidays and designdays
+                    if "holiday" in field:
+                        rule = ScheduleRule(ScheduleDay("holiday", [0], [Time(0, 0)]))
+                        holiday_schedule = rule.schedule_day
+                        rules.append(rule)
+                    if "summerdesignday" in field:
+                        rule = ScheduleRule(ScheduleDay("summerdesignday", [0], [Time(0, 0)]))
+                        summer_designday_schedule = rule.schedule_day
+                        rules.append(rule)
+                    if "winterdesignday" in field:
+                        rule = ScheduleRule(ScheduleDay("winterdesignday", [0], [Time(0, 0)]))
+                        winter_designday_schedule = rule.schedule_day
+                        rules.append(rule)
+                    if "allotherdays" in field:
+                        rule = ScheduleRule(ScheduleDay("allotherdays", [0], [Time(0, 0)]))
+                        apply_mtx = [rul.week_apply_tuple for rul in schedule_rules]
+                        if not apply_mtx:  # situation if allotherdays is the only rule.
+                            rule.apply_all = True
+                        else:
+                            for j, dow in enumerate(zip(*apply_mtx)):
+                                if not any(dow):
+                                    rule.apply_day_by_dow(j + 1)
+                        rules.append(rule)
+
+                    for rule in rules:
+                        # for each created rule in this `for` field, add rules to
+                        # ScheduleRuleset list of rules and set start_date and end_date.
+                        if len(rule.days_applied) != 0:
+                            schedule_rules.append(rule)
+
+                        # set range for rule (from previous `through` field)
+                        rule.start_date = start_date
+                        rule.end_date = end_date
+                elif "until" in field:
+                    _, hour, min = field.split(":")  # get hour and minutes
+
+                    # value is applied `until` a certain Time, but `ScheduleDay` is
+                    # applied `from` a certain Time. Also, Time is 0:23 Hours while IDF
+                    # is 1:24 Hours.
```

Comment:
It's nice to know that I am not the only one who struggles with this convention. Why in the world EnergyPlus couldn't obey how time works, I do not know. But, from the number of thumbs up on [this issue](https://github.com/NREL/EnergyPlus/issues/5360), hopefully future developers may be able to avoid it.